forch/faucetizer.py

```diff
@@ -11,132 +11,72 @@
 
     def process_faucet_config(self, faucet_config):
         """Process faucet config when base faucet config changes"""
         self._base_faucet_config = faucet_config
-        self._out_faucet_config = faucet_config
         self._faucetize()
 
-    def _faucetize(self, macs=None):
-        if not self._out_faucet_config:
-            return
+    def _faucetize(self):
+        if not self._base_faucet_config:
+            raise Exception("Base faucet configuration not provided")
 
-        if not macs:
-            macs = self._devices.keys()
-
-        commit = False
-        out_faucet_config = self._out_faucet_config
-
-        for mac in macs:
-            device = self._devices[mac]
-
-            if device.is_dirty():
-                if self._reset_port_config(device, out_faucet_config):
-                    commit = True
-
-            if device.is_complete():
-                switch = device.new_learning.switch
-                port = device.new_learning.port
+        out_faucet_config = copy.deepcopy(self._base_faucet_config)
+        for mac, device in self._devices.items():
+            if device.learning and device.behavior:
+                switch = device.learning.switch
+                port = device.learning.port
                 switch_config = out_faucet_config.get('dps', {}).get(switch, {})
                 port_config = switch_config.get('interfaces', {}).get(port)
+
                 if not port_config:
                     LOGGER.warning('Switch or port not defined in faucet config: %s %s',
                                    switch, port)
                     continue
+
                 port_config['native_vlan'] = device.behavior.vid
                 role = device.behavior.role
                 if role:
                     port_config['acls_in'] = [f'role_{role}']
-                commit = True
 
-        if commit:
-            self._commit_faucet_config(macs, out_faucet_config)
+        self._generate_faucet_config(out_faucet_config)
 
-    def _reset_port_config(self, device, out_faucet_config):
-        switch = device.old_learning.switch
-        port = device.old_learning.port
-        switch_config = out_faucet_config.get('dps', {}).get(switch, {})
-        port_config = switch_config.get('interfaces', {}).get(port)
-        if not port_config:
-            LOGGER.warning('Switch or port not defined in faucet config: %s %s', switch, port)
-            return False
-
-        base_switch_config = self._base_faucet_config.get('dps', {}).get(switch, {})
-        base_port_config = base_switch_config.get('interfaces', {}).get(port)
-        if not base_port_config:
-            LOGGER.warning('Switch or port not defined in base faucet config: %s %s', switch, port)
-            return False
-
-        port_config.update(base_port_config)
-
-        return True
-
-    def _commit_faucet_config(self, macs, out_faucet_config):
+    def _generate_faucet_config(self, out_faucet_config):
```

Comment:
To keep the class consistent, this should be a top-level (not class) function write_faucet_config as a peer to load_faucet_config.  Then, this class should just have a method that's something like "get_dynamic_faucet_config" or something, then the caller would do something like write_faucet_config(OUTPUT_PATH, FAUCITIZER.get_dynamic_faucet_config()) -- right now, you're mixing things up because the input is passed as PBs (read externally), but the output is direct to a file -- just want to be consistent in terms of the class API boundary.  The logic shouldn't change -- just instead of writing to a file, it saves it to a class variable (which is then just returned when necessary).

Reply:
Done